framework/test/unit/node/consensus/certificate_generation/commit_pool.spec.ts

```diff
@@ -335,7 +336,149 @@ describe('CommitPool', () => {
+			commitPool['_aggregateSingleCommits'] = jest.fn();
+			apiContext = createTransientAPIContext({});
+
+			bftAPI.getBFTHeights.mockResolvedValue({
+				maxHeightCertified,
+				maxHeightPrecommitted,
+			});
+
+			bftAPI.getNextHeightBFTParameters.mockResolvedValue(heightNextBFTParameters);
+
+			bftAPI.getBFTParameters.mockResolvedValue({
+				certificateThreshold: threshold,
+				validators: [
+					{ address: validatorInfo1.address, bftWeight: BigInt(1) },
+					{ address: validatorInfo2.address, bftWeight: BigInt(1) },
+				],
+			});
+		});
+
+		it('should call bft api getBFTHeights', async () => {
+			// Act
+			await commitPool['_selectAggregateCommit'](apiContext);
+
+			// Assert
+			expect(commitPool['_bftAPI'].getBFTHeights).toHaveBeenCalledWith(apiContext);
+		});
+
+		it('should call bft api getNextHeightBFTParameters', async () => {
+			// Act
+			await commitPool['_selectAggregateCommit'](apiContext);
+
+			// Assert
+			expect(commitPool['_bftAPI'].getNextHeightBFTParameters).toHaveBeenCalledWith(
+				apiContext,
+				maxHeightCertified + 1,
+			);
+		});
+
+		it('should call bft api getBFTParameters with min(heightNextBFTParameters - 1, maxHeightPrecommitted)', async () => {
+			// Act
+			await commitPool['_selectAggregateCommit'](apiContext);
+
+			// Assert
+			expect(commitPool['_bftAPI'].getBFTParameters).toHaveBeenCalledWith(
+				apiContext,
+				Math.min(heightNextBFTParameters - 1, maxHeightPrecommitted),
+			);
+		});
+
+		it('should call getBFTParameters with maxHeightPrecommitted if getNextHeightBFTParameters does not return a valid height', async () => {
+			// Arrange
+			bftAPI.getNextHeightBFTParameters.mockRejectedValue(new BFTParameterNotFoundError('Error'));
+
+			// Act
+			await commitPool['_selectAggregateCommit'](apiContext);
+
+			// Assert
+			expect(commitPool['_bftAPI'].getBFTParameters).toHaveBeenCalledWith(
+				apiContext,
+				maxHeightPrecommitted,
+			);
+		});
+
+		it('should call aggregateSingleCommits when it reaches threshold', async () => {
+			// Act
+			await commitPool['_selectAggregateCommit'](apiContext);
+
+			// Assert
+			expect(commitPool['_aggregateSingleCommits']).toHaveBeenCalledWith([singleCommit2]);
+		});
+
+		it('should not call aggregateSingleCommits when it does not reach threshold and return default aggregateCommit', async () => {
+			// Arrange
+			bftAPI.getBFTParameters.mockReturnValue({
+				certificateThreshold: 10,
+				validators: [
+					{ address: validatorInfo1.address, bftWeight: BigInt(1) },
+					{ address: validatorInfo2.address, bftWeight: BigInt(1) },
+				],
+			});
+
+			// Act
+			const result = await commitPool['_selectAggregateCommit'](apiContext);
+
+			// Assert
+			expect(commitPool['_aggregateSingleCommits']).not.toHaveBeenCalled();
+			expect(result).toEqual({
+				height: maxHeightCertified,
+				aggregationBits: Buffer.alloc(0),
+				certificateSignature: Buffer.alloc(0),
+			});
+		});
```

Comment:
these tests need to be re-reviewed with the src change

Reply:
the tests are still valid as the previous code flow actually detected when `aggregateSingleCommits` reached threshold sooner. with the new code flow it detects it after the for loop. either way the test is still valid.